distributed/tests/test_client.py

```diff
@@ -3599,6 +3599,38 @@ async def test_bad_tasks_fail(c, s, a, b):
     await asyncio.gather(a.close(), b.close())
 
 
+@gen_cluster(
+    client=True,
+    Worker=Nanny,
+    clean_kwargs={"threads": False, "processes": False},
+)
+async def test_suspicious_workers(c, s, a, b):
+    """
+    c: Client
+    s: Scheduler
+    a: Worker # name: 0
+    b: Worker # name: 1
+    """
+    # Create a future on a worker that will fail, on worker a
+    f = c.submit(sys.exit, 0)  # f: Future
+    # logger is a context manager that captures logs i.e. the errors that we will inspect.
+    with captured_logger("distributed.scheduler") as logger:
+        # when we await the future, it will fail and raise a KilledWorker error
+        with pytest.raises(KilledWorker) as info:
+            await f  # wait for the result of the future, which will fail.
+
+    logger_text = logger.getvalue()  # get the logs from the logger
+    assert f.key in logger_text  # check that the key of the future is in the logs
```

Comment:
I think we can drop the assertion on the logs. In this case we only care about the _event_ being emitted but not much about the log.

In fact, this test is very similar to `test_bad_tasks_fail` and it might make sense to merge the two tests into one. I think if you add the assert statment below to `test_bad_tasks_fail` we'll get the same coverage but only have to run this expensive/slow test once

Reply:
I'll move this into `test_bad_tasks_fail` once this is passing tests.